flink-runtime/src/main/java/org/apache/flink/runtime/operators/util/JoinTaskIterator.java

```diff
@@ -68,4 +72,9 @@
 	 * method signals an interrupt to that procedure.
 	 */
 	void abort();
+
+	/**
+	 * Allows to reset the iterator
+	 */
+	void reset(MutableObjectIterator<V1> in1, MutableObjectIterator<V2> in2, TypeSerializer<V1> serializer1, TypeSerializer<V2> serializer2, TypeComparator<V1> comp1, TypeComparator<V2> comp2, TypePairComparatorFactory<V1, V2> pairComparatorFactory);
```

Comment:
Could you pass in only those objects here, which might change during a reset? I think these should only be the two inputs.